src/darwin/Framework/CHIP/MTRDeviceControllerDelegate.h

```diff
@@ -23,7 +23,6 @@ typedef NS_ENUM(NSInteger, MTRCommissioningStatus) {
     MTRCommissioningStatusUnknown = 0,
     MTRCommissioningStatusSuccess = 1,
     MTRCommissioningStatusFailed = 2,
-    MTRCommissioningStatusDiscoveringMoreDevices = 3
```

Comment:
We can't just remove this; that breaks source compat.  Please mark this deprecated, indicating that it never actually happens, instead?

Reply:
I am a bit confused about which deprecated mark I should used in #25176. For example if I change it to:
 typedef NS_ENUM(NSInteger, MTRCommissioningStatus) {
     MTRCommissioningStatusUnknown = 0,
     MTRCommissioningStatusSuccess = 1,
     MTRCommissioningStatusFailed = 2,
-    MTRCommissioningStatusDiscoveringMoreDevices = 3
+    MTRCommissioningStatusDiscoveringMoreDevices API_DEPRECATED(
+        "MTRCommissioningStatusDiscoveringMoreDevices is not used.", ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
+    = 3,
 } API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4));
It does not seems to make sense. My understanding of that is:
“This API is introduced in ios(16.4), macos(13.3), watchos(9.4), tvos(16.4) but MTRCommissioningStatusDiscoveringMoreDevices is deprecated in in ios(16.4), macos(13.3), watchos(9.4), tvos(16.4)